typescript/cli/src/rebalancer/strategy/BaseStrategy.ts

```diff
@@ -16,28 +17,30 @@ export type Delta = { chain: ChainName; amount: bigint };
 export abstract class BaseStrategy implements IStrategy {
   protected readonly chains: ChainName[];
   protected readonly metrics?: Metrics;
+  protected readonly logger: Logger;
 
-  constructor(chains: ChainName[], metrics?: Metrics) {
+  constructor(chains: ChainName[], logger: Logger, metrics?: Metrics) {
```

Comment:
nit: why don't you change the constructor to:
```typescript
constructor(
    protected readonly chains: ChainName[],
    protected readonly logger: Logger,  
    protected readonly metrics?: Metrics,
) {}
```

this way you can avoid having to assign in the constructor body